### PR TITLE
Use default provider when using selected text only

### PR DESCRIPTION
**Pull request by jahands**
Cleaner solution for #17 than #19

### CODE REVIEW

**Comment by SrTobi** [comment]
year, that was also my first idea. the problem is, that if `noInputBoxIfTextSelected` is set to false, then this solution won't work.

**Comment by jahands** [comment]
I think by default selections should use provider code if `noInputBoxIfTextSelected` is disabled.

I'd think searching github or stackoverflow would be a common use-case for programmers.

Perhaps an option like `alwaysUseDefaultForSelection` should be added?

**Comment by SrTobi** [comment]
ok, lets discuss the different cases:
(1) no selection, user wants to use default provider
=> that works. he simply enters his query and thats it. if the first word of his query is also a provider-shortcut he is unlucky and has to type the search provider explicitly as well (or he has defined useDefaultProviderOnly = true). thats ok.
(2) no selection, user wants to use specific provider
=> even simpler. he simply enters the shortcut and his query and tada it works.
(3) selected text, user wants to use specific provider
=> simple. he selects his text, activates the extension and in the input box he prepends the specific shortcut 
(4) selected text, user wants to use default provider
=> it is simple if useDefaultProviderOnly = true, then simply use the default provider (even no input box)
=> the problem is if useDefaultProviderOnly = false. we can not simply search the output from the input box (^^) for a shortcut. we have to check if he added a shortcut or not. And it is easiest if we simply compare it to the original selection. And i do not see an advantage in adding a new very specific option.

**Comment by jahands** [comment]
I'm not sure I understand what you're trying to say on 4...

**Comment by jahands** [comment]
The new option is so that
=> Use default provider for selection searches
is separate from
=> Use default provider for both types of searches

**Comment by jahands** [comment]
Wait....

**Comment by jahands** [comment]
`codebing.useDefaultProviderOnly`
=> Always use default provider for everything

`codebing.noInputBoxIfTextSelected`
=> Insta search selection using default provider

`codebing.alwaysUseDefaultForSelection`
=> use default provider even if `codebing.noInputBoxIfTextSelected` is set to `false`

Someone may find it common to edit selection queries before searching, but doesn't want to worry about if the beginning of the query contains a provider code or not.

File: src/utils.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
-
-
-export function isNullOrEmpty(s:string):boolean {
+export function isNullOrEmpty(s: string): boolean {
 	return s == null || s == "";
-}
\ No newline at end of file
+}
+export function startsWith(s: string, startsWith: string) {
+	return s != null && startsWith != null && s.indexOf(startsWith) == 0
```
Comment (SrTobi) [inline src/utils.ts]: missing `;` ^^

**Pull request merged**